tests/utils_tests/test_dateparse.py

```diff
@@ -108,6 +108,11 @@ def test_fractions_of_seconds(self):
 
     def test_negative(self):
         self.assertEqual(parse_duration('-4 15:30'), timedelta(days=-4, minutes=15, seconds=30))
+        self.assertEqual(parse_duration('4 -15:30'), timedelta(days=4, minutes=-15, seconds=30))
+        self.assertEqual(parse_duration('4 15:-30'), timedelta(days=4, minutes=15, seconds=-30))
```

Comment:
This doesn't look like correct notation. What about the approach in https://github.com/django/django/pull/7814? I think you need to test this in the context of `DurationField` and SQLite as described on the ticket.